diff --git a/node_modules/react-native-qrcode-scanner-view/QRScanner.js b/node_modules/react-native-qrcode-scanner-view/QRScanner.js
index 3439244..a51e900 100644
--- a/node_modules/react-native-qrcode-scanner-view/QRScanner.js
+++ b/node_modules/react-native-qrcode-scanner-view/QRScanner.js
@@ -286,14 +286,17 @@ export default class QRScannerView extends Component {
     super(props);
     // 避免频繁触发扫描回调
     this.onScanResult = throttle(this.onScanResult, this.props.scanInterval, { maxWait: 0, trailing: false });
+    this.listener=null;
   }
   
+  
   componentDidMount(){
-    AppState.addEventListener('change', this.handleAppStateChange);
+    this.listener=AppState.addEventListener('change', this.handleAppStateChange);
   }
   
   componentWillUnmount(){
-    AppState.removeEventListener('change', this.handleAppStateChange);
+    // AppState.removeEventListener('change', this.handleAppStateChange);
+    this.listener.remove();
     this.rnCamera && this.rnCamera.pausePreview();
   }
   
